engine/src/main/scala/cromwell/engine/workflow/lifecycle/ShadowMaterializeWorkflowDescriptorActor.scala

```diff
@@ -0,0 +1,282 @@
+  val RuntimeBackendKey: String = "runtime"
+
+  def props(): Props = Props(new ShadowMaterializeWorkflowDescriptorActor)
+
+  /*
+  Commands
+   */
+  sealed trait ShadowMaterializeWorkflowDescriptorActorMessage
+  case class ShadowMaterializeWorkflowDescriptorCommand(id: WorkflowId,
+                                                        workflowSourceFiles: WorkflowSourceFiles,
+                                                        conf: Config) extends ShadowMaterializeWorkflowDescriptorActorMessage
+  case object ShadowMaterializeWorkflowDescriptorAbortCommand
+
+  /*
+  Responses
+   */
+  sealed trait ShadowWorkflowDescriptorMaterializationResult extends ShadowMaterializeWorkflowDescriptorActorMessage
+  case class ShadowMaterializeWorkflowDescriptorSuccessResponse(workflowDescriptor: EngineWorkflowDescriptor) extends ShadowWorkflowDescriptorMaterializationResult
+  case class ShadowMaterializeWorkflowDescriptorFailureResponse(reason: Throwable) extends Exception with ShadowWorkflowDescriptorMaterializationResult
+
+  /*
+  States
+   */
+  sealed trait ShadowMaterializeWorkflowDescriptorActorState { def terminal = false }
+  sealed trait ShadowMaterializeWorkflowDescriptorActorTerminalState extends ShadowMaterializeWorkflowDescriptorActorState {
+    override val terminal = true
+  }
+  case object ReadyToMaterializeState extends ShadowMaterializeWorkflowDescriptorActorState
+  case object MaterializationSuccessfulState extends ShadowMaterializeWorkflowDescriptorActorTerminalState
+  case object MaterializationFailedState extends ShadowMaterializeWorkflowDescriptorActorTerminalState
+  case object MaterializationAbortedState extends ShadowMaterializeWorkflowDescriptorActorTerminalState
+
+  /*
+  Data
+   */
+  case class ShadowMaterializeWorkflowDescriptorActorData()
+
+  import lenthall.config.ScalaConfig._
+
+  private val DefaultWorkflowFailureMode = NoNewCalls.toString
+
+  def workflowLogOptions(conf: Config): Option[WorkflowLogOptions] = {
+    for {
+      workflowConfig <- conf.getConfigOption("workflow-options")
+      dir <- workflowConfig.getStringOption("workflow-log-dir") if !dir.isEmpty
+      temporary <- workflowConfig.getBooleanOption("workflow-log-temporary") orElse Option(true)
+    } yield WorkflowLogOptions(Paths.get(dir), temporary)
+  }
+
+  /**
+    * Checks if a given backend name is plugged-in in to the Cromwell sub-system and returns if true.
+    * Throws an exception if `backendName` is not found in the list
+    */
+  private def validateBackend(backendName: String): String = {
+    if (PluggedInBackends.exists(_.equalsIgnoreCase(backendName))) backendName
+    else throw new Exception(s"$backendName not found in the list of plugged in backends!")
+
+  }
+}
+
+class ShadowMaterializeWorkflowDescriptorActor() extends LoggingFSM[ShadowMaterializeWorkflowDescriptorActorState, ShadowMaterializeWorkflowDescriptorActorData] with LazyLogging {
```

Comment:
Is  this needed to be an FSM? It only has one actionable state and I cannot think of any other states that can be here 😕